templates/performance_optimized.j2

```diff
@@ -17,19 +17,11 @@ op_defaults \
 # production HANA
 #
 
-{% if data.platform == "gcp" %}
 primitive rsc_ip_{{ sid }}_HDB{{ instance }} ocf:heartbeat:IPaddr2 \
     params \
-    ip="{{ data.virtual_ip }}" cidr_netmask="32" nic="eth0" \
+    ip="{{ data.virtual_ip }}" cidr_netmask="{{ 32 if data.platform == "gcp" else data.virtual_ip_mask }}" {{ "nic=" + pillar.cluster.interface if data.platform == "gcp" }} \
```

Comment:
I think we could assume "eth0" as default, independent of the platform. I think is very common to have more than one interface, so for all platforms the customer can change the virtual IP to a specific interface.

```suggestion
    ip="{{ data.virtual_ip }}" cidr_netmask="{{ 32 if data.platform == "gcp" else data.virtual_ip_mask }}" {{ "nic=" ~ pillar.cluster.interface if pillar.cluster.interface else "eth0" }} \
```

P.S.: AFAIK, the `~` operator is the way to concatenate on Jinja.

Reply:
@diegoakechi Thanks, let me try your suggested change. You are right for `~`.

Reply:
@diegoakechi @juadk 
```
{{ "nic=" ~ pillar.cluster.interface or 'eth0' }}
```
might work as well (I didn't test it)

Reply:
@diegoakechi @arbulu89 The both solutions work but if I remove `interface: eth0` from /srv/pillar/cluster.sls, templating fails with this error:
`Unable to manage file: Jinja variable 'salt.utils.odict.OrderedDict object' has no attribute 'interface'`

Maybe my syntax `pillar.cluster.interface` isn't the best way or we could consider that cluster.interface is mandatory for cluster deployment, AFAIK, it is.

Reply:
Well, it shouldn't be mandatory if we have a default =)

What if the inversion?

```suggestion
    ip="{{ data.virtual_ip }}" cidr_netmask="{{ 32 if data.platform == "gcp" else data.virtual_ip_mask }}" {{ "nic=" ~ "eth0" if not pillar.cluster.interface else pillar.cluster.interface }} \
```

Reply:
@juadk 
Other solution would be:
```
{{ "nic=" ~ pillar.cluster.interface|default('eth0') }}
```

Reply:
This is much more elegant =)

Reply:
And works better =)

Reply:
Updated and tested with GCP as well as AWS.